pkg/interfaces/handler/superlike_handler.go

```diff
@@ -0,0 +1,131 @@
+		ctx := request.Context()
+		ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+		defer cancel()
+		UID := dcontext.GetUIDFromContext(ctx)
+		superlikes, err := superlikeH.superlikeUseCase.Select(ctx, UID)
+		if err != nil {
+			log.Printf("[ERROR] failed to Select from superlike: %v", err.Error())
+			http.Error(writer, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		res := make([]*SuperlikeResponse, len(superlikes))
+		for i, l := range superlikes {
+			var ls SuperlikeResponse
+			ls.RecruitID = l.RecruitID
+			ls.CreatedAt = l.CreatedAt
+			res[i] = &ls
+		}
+		var respms SuperlikeResponses
+		respms.Superlikes = res
+		jsonresponse, err := json.Marshal(respms)
+		if err != nil {
+			log.Printf("[ERROR] failed to marshal superlikes: %v", err.Error())
+			http.Error(writer, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		writer.WriteHeader(http.StatusOK)
+		writer.Header().Set("Content-Type", "application/json")
+		writer.Write(jsonresponse)
+	}
+}
+
+type superlikeInsertRequest struct {
+	RecruitID int32 `json:"recruit_id"`
+}
+
+// HandleInsert
+func (superlikeH *superlikeHandler) HandleInsert() http.HandlerFunc {
+	return func(writer http.ResponseWriter, request *http.Request) {
+		defer request.Body.Close()
+		SuperlikeRequest := new(superlikeInsertRequest)
+		if err := json.NewDecoder(request.Body).Decode(&SuperlikeRequest); err != nil {
+			log.Printf("[ERROR] failed to JsonDecord: %v", err.Error())
+			http.Error(writer, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		// TODO::読み込んだjsonデータのバリデーションがほしい
+		ctx := request.Context()
+		ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+		defer cancel()
+		UID := dcontext.GetUIDFromContext(ctx)
+		if UID == "" {
+			log.Printf("[INFO] failed to GetUID: ")
+			http.Error(writer, "Could not get UID", http.StatusInternalServerError)
+			return
+		}
+		superlike := new(superlikeM.Superlike)
+		superlike.RecruitID = SuperlikeRequest.RecruitID
+		err := superlikeH.superlikeUseCase.Insert(ctx, superlike, UID)
+		if err != nil {
+			log.Printf("[ERROR] failed to Insert: %v", err.Error())
+			http.Error(writer, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		writer.WriteHeader(http.StatusCreated)
+	}
+}
+
+// HandleUpdate
+func (superlikeH *superlikeHandler) HandleUpdate() http.HandlerFunc {
+	return func(writer http.ResponseWriter, request *http.Request) {
+		panic("do something")
+	}
+}
+
+// HandleDelete
+func (superlikeH *superlikeHandler) HandleDelete() http.HandlerFunc {
+	return func(writer http.ResponseWriter, request *http.Request) {
+		panic("do something")
+	}
+}
+
+// SuperlikeRequest
+type SuperlikeRequest struct { // nolint
+	// Need to implement field
+}
+
+type SuperlikeResponses struct {
+	Superlikes []*SuperlikeResponse `json:"superlikes"`
+}
+
+// SuperlikeResponse ...
+type SuperlikeResponse struct {
+	RecruitID int32     `json:"recruit_id"`
+	CreatedAt time.Time `json:"created_at"`
+}
```

Comment:
```suggestion
type superlikeInsertResponses struct {
	Superlikes []*superlikeInsertResponse `json:"superlikes"`
}

// superlikeResponse ...
type superlikeInsertResponse struct {
	RecruitID int32     `json:"recruit_id"`
	CreatedAt time.Time `json:"created_at"`
}
```
requestはunexportedなのにresponseはexported？

Reply:
ごめんどちらもunexportedがいいわ、